js_tests/admin/DateTimeShortcuts.test.js

```diff
@@ -40,3 +40,15 @@ QUnit.test('time zone offset warning', function(assert) {
     $('body').attr('data-admin-utc-offset', savedOffset);
     assert.equal($('.timezonewarning').text(), 'Note: You are 1 hour behind server time.');
 });
+
+QUnit.test('Calendar regular expression callback', function(assert) {
+    const $ = django.jQuery;
+    let format = "\n February \n 12 \n 1998 \r \r \r \t \t \t \\ \\ ' '";
+    format = format.replace(/\\/g, "\\\\");
+    format = format.replace(new RegExp('\r', 'g'), '\\r')
+            .replace(new RegExp('\n', 'g'), '\\n')
+            .replace(new RegExp('\t', 'g'), '\\t')
+            .replace(new RegExp("'", 'g'), "\\'")
+            ;
```

Comment:
We shouldn't duplicate the code from the actual function, but instead be able to use the code from `django/contrib/admin/static/admin/js/admin/DateTimeShortcuts.js`

We could add a function that will specifically escape the date format and call it here:
```
const format = "\n %Y \r %m \r \t %d \t %H \\ %M \\ '%S'\n";
const escapedFormat = DateTimeShortcuts.escapeFormat(format);
assert.equal(escapedFormat, "\\n %Y \\r %m \\r \\t %d \\t %H \\\\ %M \\\\ \\'%S\\'\n");
```
Something in this vein.

Reply:
Ok, thank you I will do that.

Reply:
I have done as you suggested, escapeFormat is now a separate function.